src/tomato/ketchup/functions.py

```diff
@@ -153,66 +167,137 @@ def status(args: Namespace) -> None:
+            rows = [
+                (pip,) + dbhandler.pipeline_get_info(state["path"], pip, state["type"])
+                for pip in pips
+            ]
+            columns = list(map(list, zip(*rows)))
+            column_names = ["pipeline", "sampleid", "ready", "jobid", "pid"]
+            data = {name: column for name, column in zip(column_names, columns)}
+            print(json.dumps(data))
+        elif "queue" in args.jobid:
+            jobs = dbhandler.job_get_all(queue["path"], type=queue["type"])
+            running = dbhandler.pipeline_get_running(state["path"], type=state["type"])
+            rows = []
+            jobids = []
+            column_names = ["jobid", "jobname", "status", "pipeline", "pid"]
+            for jobid, jobname, payload, status in jobs:
+                if not status.startswith("c") or args.verbose - args.quiet > 0:
+                    row = [jobid, jobname, status, None, None]
+                    jobids += [jobid]
+                    rows += [row]
+            for pip, pjobid, pid in running:
+                try:
+                    idx = jobids.index(pjobid)
+                    rows[idx][3:4] = [pip, pid]
+                except ValueError:
+                    pass  # not in running
+            columns = list(map(list, zip(*rows)))
+            data = {name: column for name, column in zip(column_names, columns)}
+            print(json.dumps(data))
+        else:
+            column_names = [
+                "jobid",
+                "jobname",
+                "status",
+                "submitted_at",
+                "executed at",
+                "completed_at",
+                "pipeline",
+                "pid",
+            ]
+            running = dbhandler.pipeline_get_running(state["path"], type=state["type"])
+            if running:
+                pips, pjobids, pids = zip(*running)
+            rows = []
+            for i, jobid in enumerate(args.jobid):
+                ji = dbhandler.job_get_info(queue["path"], jobid, type=queue["type"])
+                jobname, payload, status, submitted_at, executed_at, completed_at = ji
+                row = [
+                    int(jobid),
+                    jobname,
+                    status,
+                    submitted_at,
+                    executed_at,
+                    completed_at,
+                    None,
+                    None,
+                ]
+                if running and (status.startswith("r") or status.startswith("c")):
+                    try:
+                        i = pjobids.index(int(jobid))
+                        row[6:7] = [pips[i], pids[i]]
+                    except ValueError:
+                        pass  # not in running
+                rows += [row]
+            columns = list(map(list, zip(*rows)))
+            data = {name: column for name, column in zip(column_names, columns)}
+            print(json.dumps(data))
```

Comment:
It'd perhaps be better to keep the output between `-J` and without consistent (i.e. print the same information). So maybe this row-parsing logic could be moved into a separate function and called by both branches of the code, with the only difference being the formatting?

Reply:
I was scared to touch the other status printing because I thought it might be used and parsed in Aiida, but I don't think that is the case. If not I can use the same logic on both.

Reply:
Updated to use the same logic, this leads to it printing slightly differently than before

ketchup status
```
pipeline             sampleid                  ready jobid  pid
======================================================================
dummy-10             dummy_random_2_0.1        0     None   None
dummy-5              dummy_random_1_0.1        0     None   None
```
ketchup status -J
```
{"pipeline": ["dummy-10", "dummy-5"], "sampleid": ["dummy_random_2_0.1", "dummy_random_1_0.1"], "ready": [0, 0], "jobid": [null, null], "pid": [null, null]}
```
ketchup status queue -v
```
jobid  jobname         status pipeline             pid
=============================================================
1      None            cd     None                 None
2      None            cd     None                 None
```
ketchup status queue -v -J
```
{"jobid": [1, 2], "jobname": [null, null], "status": ["cd", "cd"], "pipeline": [null, null], "pid": [null, null]}
```
ketchup status 1
```
jobid: 1
jobname: None
status: cd
submitted_at: 2024-06-03 12:58:15.114654+00:00
executed_at: 2024-06-03 13:26:26.278559+00:00
completed_at: 2024-06-03 13:26:30.494332+00:00
pipeline: None
pid: None
```
ketchup status 1 -J
```
{"jobid": [1], "jobname": [null], "status": ["cd"], "submitted_at": ["2024-06-03 12:58:15.114654+00:00"], "executed_at": ["2024-06-03 13:26:26.278559+00:00"], "completed_at": ["2024-06-03 13:26:30.494332+00:00"], "pipeline": [null], "pid": [null]}
```

Reply:
@edan-bainglass: would you mind double checking that this doesn't screw up anything on the AiiDA side? I believe the individual `ketchup status <jobid>` were parsed as yaml anyway, and you didn't really use the table. But better safe than sorry!

Reply:
I've changed it to yaml dump the data, it should be the same format as before (but will print null values)

```
(.venv) C:\tomato>ketchup status 1 2
- jobid: 1
  jobname: null
  status: cd
  submitted_at: '2024-06-03 12:58:15.114654+00:00'
  executed_at: '2024-06-03 13:26:26.278559+00:00'
  completed_at: '2024-06-03 13:26:30.494332+00:00'
  pipeline: null
  pid: null
- jobid: 2
  jobname: null
  status: cd
  submitted_at: '2024-06-03 13:27:44.258318+00:00'
  executed_at: '2024-06-03 13:27:45.524514+00:00'
  completed_at: '2024-06-03 13:27:49.291359+00:00'
  pipeline: null
  pid: null
  ```